### PR TITLE
Docs: format sys.float_info properly

### PR DESCRIPTION
**Pull request by erlend-aasland**
- Normalise capitalisation and punctuation
- Use .. attribute:: markup for named tuple attributes
- Use :c:macro: markup for C macros
- Use a list for the 'rounds' attribute values
- Use list-table, for better .rst readability

----
:books: Documentation preview :books:: https://cpython-previews--108107.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I really like the `list-table` markup.

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -574,57 +574,78 @@ always available.
...
-   |                     |                     | implementation-defined rounding behavior.        |
-   +---------------------+---------------------+--------------------------------------------------+
+   .. list-table:: Attributes of the :data:`!float_info` :term:`named tuple`
+      :header-rows: 1
+
+      * - attribute
+        - float.h macro
+        - explanation
+
+      * - :attr:`!epsilon`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.epsilon
```

File: Doc/library/sys.rst [hunk 2]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_EPSILON`
+        - difference between 1.0 and the least value greater than 1.0 that is
+          representable as a float.
+
+          See also :func:`math.ulp`.
+
+      * - :attr:`!dig`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.dig
```

File: Doc/library/sys.rst [hunk 3]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_DIG`
+        - The maximum number of decimal digits that can be faithfully
+          represented in a float; see below.
+
+      * - :attr:`!mant_dig`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.mant_dig
```

File: Doc/library/sys.rst [hunk 4]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MANT_DIG`
+        - Float precision: the number of base-``radix`` digits in the
+          significand of a float.
+
+      * - :attr:`!max`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.max
```

File: Doc/library/sys.rst [hunk 5]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MAX`
+        - The maximum representable positive finite float.
+
+      * - :attr:`!max_exp`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.max_exp
```

File: Doc/library/sys.rst [hunk 6]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MAX_EXP`
+        - The maximum integer *e* such that ``radix**(e-1)`` is a representable
+          finite float.
+
+      * - :attr:`!max_10_exp`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.max_10_exp
```

File: Doc/library/sys.rst [hunk 7]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MAX_10_EXP`
+        - The maximum integer *e* such that ``10**e`` is in the range of
+          representable finite floats.
+
+      * - :attr:`!min`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.min
```

File: Doc/library/sys.rst [hunk 8]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MIN`
+        - The minimum representable positive *normalized* float.
+
+          Use :func:`math.ulp(0.0) <math.ulp>` to get the smallest positive
+          *denormalized* representable float.
+
+      * - :attr:`!min_exp`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.min_exp
```

File: Doc/library/sys.rst [hunk 9]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MIN_EXP`
+        - The minimum integer *e* such that ``radix**(e-1)`` is a normalized
+          float.
+
+      * - :attr:`!min_10_exp`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.min_10_exp
```

File: Doc/library/sys.rst [hunk 10]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!DBL_MIN_10_EXP`
+        - The minimum integer *e* such that ``10**e`` is a normalized float.
+
+      * - :attr:`!radix`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.radix
```

File: Doc/library/sys.rst [hunk 11]
```diff
@@ -574,57 +574,78 @@ always available.
...
+        - :c:macro:`!FLT_RADIX`
+        - The radix of exponent representation.
+
+      * - :attr:`!rounds`
```
Comment (AA-Turner) [inline Doc/library/sys.rst]:
```suggestion
      * - .. attribute:: float_info.rounds
```

**Comment by rhettinger** [comment]
Perhaps also mention subnormals because this was a point of confusion in the past: https://github.com/python/cpython/issues/93488

**Comment by rhettinger** [comment]
Also see the comments and edits in https://github.com/python/cpython/issues/85988

**Comment by AA-Turner** [comment]
> Perhaps also mention subnormals

This PR is just about re-formatting the table such that it reads better, so I think adding any major new content would be out-of-scope? Will defer to Erlend, though.

A

**Comment by mdickinson** [comment]
+1 for the formatting changes; agreed that we can tweak the content if necessary in a separate PR.

**Review by mdickinson — APPROVED** [review]

**Comment by erlend-aasland** [comment]
> > Perhaps also mention subnormals
> 
> This PR is just about re-formatting the table such that it reads better, so I think adding any major new content would be out-of-scope? Will defer to Erlend, though.

Definitely; content changes are out of scope for this PR.

I'll add your Sphinx suggestions before landing.

Thanks for chiming in, y'all.

**Pull request merged**